python/ngen_cal/src/ngen/cal/_hookspec.py

```diff
@@ -46,20 +46,41 @@ def ngen_cal_finish(exception: Exception | None) -> None:
     `exception` will be non-none if an exception was raised during calibration.
     """
 
+
 class ModelHooks:
     @hookspec(firstresult=True)
-    def ngen_cal_model_output(self, id: str | None) -> Series:
+    def ngen_cal_model_observations(
+        self,
+        id: str,
+        start_time: datetime,
+        end_time: datetime,
+        simulation_interval: pd.Timedelta,
+    ) -> pd.Series:
         """
-            Called during each calibration iteration to provide the model output in the form
-            of a pandas Series, indexed by time.
-            Output series should be in units of cubic meters per second.
+        Called during each calibration iteration to provide truth / observation
+        values in the form of a pandas Series, indexed by time with a record
+        every `simulation_interval`.
+        The returned pandas Series should be in units of cubic meters per
+        second.
+
+        `id`: USGS gage station id
```

Comment:
Great questions! Glad to hear that we are thinking on the same page. Two thoughts preface with: I went with this design b.c. I decided to just keep it as simple as possible for now. Since we are pre-`1.0`, I figured we might change this as we learn things along the way.

1. I'd originally written up a complex object that encapsulates what Mike's group refers to a point-of-interest. Like you are saying, the thinking being that the `id` for the model data is likely not the same `id` for the observation (truth) data. I decided against that to err on the side of simplicity for now. I am open changing that now or in the future.
2. I also though about passing a `Nexus`, but thinking back to previous conversations  we'd had about keeping things model agnostic, it seemed passing a `NextGen` specific idea went against that. If we ever do go in the direction of something like a POI interface as a solution to this problem, I figured we could always create a type that implements the POI interface for a `Nexus` for example.

Reply:
Conversation with @hellkite500, in theory a `HydroLocation` or `Nexus` makes the most sense here (really a `Nexus` b.c. its more general). We have decided to go with `Nexus` for now.

Reply:
I think the `Nexus` is the right way to go for now, it encapsulates a (eventually) list of `hydrolocation` which can reference different entities.

Reply:
Related: https://github.com/NOAA-OWP/hypy/issues/37